### PR TITLE
Fix ability to have duplicate account names

### LINKED ISSUE #316: Duplicate Account names should not be possible.

### PR DESCRIPTION
**Pull request by qedi-r**
Fixes #316 

I took a look at PR #2776, and I think I addressed all the original comments there. I tried to keep it consistently styled as well.

Create account:
image

Desktop account page:
image

Mobile account page:
image

### CODE REVIEW

File: packages/desktop-client/src/components/modals/CreateLocalAccountModal.tsx [hunk 1]
```diff
@@ -24,24 +27,51 @@
...
+  const accounts = useAccounts.useAccounts();
   const [name, setName] = useState('');
   const [offbudget, setOffbudget] = useState(false);
   const [balance, setBalance] = useState('0');
 
-  const [nameError, setNameError] = useState(false);
+  const [nameError, setNameError] = useState(null);
   const [balanceError, setBalanceError] = useState(false);
 
   const validateBalance = balance => !isNaN(parseFloat(balance));
```
[the remark this answers is not in the document]
Comment (qedi-r) [reply packages/desktop-client/src/components/modals/CreateLocalAccountModal.tsx:47]: Validate balance is outside of scope, rabbit.

File: packages/desktop-client/src/components/modals/CreateLocalAccountModal.tsx [hunk 2]
```diff
@@ -24,24 +27,51 @@
 import { Text } from '../common/Text';
 import { View } from '../common/View';
 import { Checkbox } from '../forms';
+import { validateAccountName } from '../util/accountValidation';
+
+function accountNameIsInUse(accountName: string, accounts: AccountEntity[]) {
+  return accounts.map(a => a.name).includes(accountName);
```
[the remark this answers is not in the document]
Comment (qedi-r) [reply packages/desktop-client/src/components/modals/CreateLocalAccountModal.tsx:33]: I kinda though sensitivity would be valid uniqueness, but the AI disagrees, apparently. Maintainer thoughts?

File: packages/desktop-client/src/components/accounts/Header.jsx [hunk 3]
```diff
@@ -2,5 +1,8 @@
+import React, { useState, useRef, Fragment } from 'react';
 import { useHotkeys } from 'react-hotkeys-hook';
-import { Trans, useTranslation } from 'react-i18next';
+import { Trans } from 'react-i18next';
+
+import { t } from 'i18next';
```
[the remark this answers is not in the document]
Comment (qedi-r) [reply packages/desktop-client/src/components/accounts/Header.jsx]: I was going to ask about this one - it looks like it's in use like this in other places in the product, so I thought this was acceptable. Maybe clarifying statements should be made on https://actualbudget.org/docs/contributing/i18n/?

File: packages/desktop-client/src/components/modals/CreateLocalAccountModal.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (qedi-r) [reply packages/desktop-client/src/components/modals/CreateLocalAccountModal.tsx]: (Though, this was made on the one that isn't in use.)

File: packages/desktop-client/src/components/accounts/Header.jsx [hunk 4]
```diff
@@ -176,99 +177,21 @@ export function AccountHeader({
...
-                      ? theme.sidebarItemBackgroundFailed
-                      : theme.sidebarItemBackgroundPositive,
-                  marginRight: '4px',
-                  width: 8,
-                  height: 8,
-                  borderRadius: 8,
-                }}
+              <AccountSyncSidebar
+                account={account}
+                failedAccount={failedAccounts}
```
[the remark this answers is not in the document]
Comment (qedi-r) [reply packages/desktop-client/src/components/accounts/Header.jsx]: Useful catch, rabbit.

**Comment by youngcw** [comment]
Instead of saying "must be unique" it probably would be better to say "name already exists". That would be more clear.

**Comment by qedi-r** [comment]
> Instead of saying "must be unique" it probably would be better to say "name already exists". That would be more clear.

Updated.

**Comment by youngcw** [comment]
This looks good.

Can I get @jfdoming or @matt-fidd or @joel-jeremy to check the translations in here?

File: packages/desktop-client/src/components/accounts/Header.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jfdoming) [reply packages/desktop-client/src/components/accounts/Header.jsx:5]: I would suggest keeping the `useTranslation` here, there is extra logic to make it well-behaved on language changes and such. We really should use it everywhere but one step at a time.

File: packages/desktop-client/src/components/modals/AccountMenuModal.tsx [hunk 5]
```diff
@@ -1,8 +1,11 @@
-import React, { type ComponentProps, useRef, useState } from 'react';
+import React, { type ComponentProps, Fragment, useRef, useState } from 'react';
+
+import { t } from 'i18next';
```
Comment (jfdoming) [inline packages/desktop-client/src/components/modals/AccountMenuModal.tsx:4]: Same comment here

**Review by jfdoming — COMMENTED** [review]
Looks good, just one comment

File: packages/desktop-client/src/components/modals/AccountMenuModal.tsx (hunk 5 above)
Comment (qedi-r) [reply packages/desktop-client/src/components/modals/AccountMenuModal.tsx:4]: Okay. I just was applying it, assuming it was the same given it's use in the codebase, though now after reading, I think I understand the differences more.

Comment (qedi-r) [reply packages/desktop-client/src/components/modals/AccountMenuModal.tsx:3]: Both are updated.

**Review by jfdoming — COMMENTED** [review]
Looks good now! Though I see some tests are failing

**Review by youngcw — APPROVED** [review]

**Pull request merged**